From 1852d912a0d946244459bbeaa3cb3ea618081bb9 Mon Sep 17 00:00:00 2001
From: Jonathon Anderson <anderson.jonathonm@gmail.com>
Date: Mon, 13 Oct 2025 15:06:45 -0500
Subject: [PATCH] Add #include <cstdint>

This patch is a combination of https://github.com/dyninst/dyninst/pull/1880
(which fixes sha1.C) and `github-pr1880-ish.patch` from the Fedora packaging
for Dyninst (https://src.fedoraproject.org/rpms/dyninst/blob/rawhide/f/github-pr1880-ish.patch).

Signed-off-by: Jonathon Anderson <anderson.jonathonm@gmail.com>
---
 common/h/Annotatable.h                    | 1 +
 common/src/arch-x86.h                     | 1 +
 common/src/sha1.C                         | 1 +
 instructionAPI/h/ArchSpecificFormatters.h | 1 +
 4 files changed, 4 insertions(+)

diff --git a/common/h/Annotatable.h b/common/h/Annotatable.h
index b5bc5cea1..b1c06b7b8 100644
--- a/common/h/Annotatable.h
+++ b/common/h/Annotatable.h
@@ -44,6 +44,7 @@
 #include <assert.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <cstdint>
 #include "util.h"
 #include "compiler_annotations.h"
 
diff --git a/common/src/arch-x86.h b/common/src/arch-x86.h
index ecdc3dc13..7745306d3 100644
--- a/common/src/arch-x86.h
+++ b/common/src/arch-x86.h
@@ -40,6 +40,7 @@
 #include <set>
 #include <map>
 #include <vector>
+#include <cstdint>
 #include "entryIDs.h"
 #include "registers/MachRegister.h"
 #include "common/src/ia32_locations.h"
diff --git a/common/src/sha1.C b/common/src/sha1.C
index 08dd9f6bb..489e6314a 100644
--- a/common/src/sha1.C
+++ b/common/src/sha1.C
@@ -102,6 +102,7 @@ A million repetitions of "a"
 
 #include <stdio.h>
 #include <string.h>
+#include <cstdint>
 
 #include "dyntypes.h"
 #include "common/src/sha1.h"
diff --git a/instructionAPI/h/ArchSpecificFormatters.h b/instructionAPI/h/ArchSpecificFormatters.h
index b32d4cc14..145c604f9 100644
--- a/instructionAPI/h/ArchSpecificFormatters.h
+++ b/instructionAPI/h/ArchSpecificFormatters.h
@@ -34,6 +34,7 @@
 #include <map>
 #include <vector>
 #include <string>
+#include <cstdint>
 #include "Architecture.h"
 #include "registers/MachRegister.h"
 
-- 
2.51.0

